### PR TITLE
S32K1XX FlexCAN fix & EEEPROM driver

### PR DESCRIPTION
**Pull request by PetervdPerk-NXP**
This PR includes contains some fixes for the FlexCAN driver:
- Bitrate calculation fix
- FlexCAN TX done and transmit race condition

Furthermore it includes the Enhanced EEPROM (4kb) block device driver which allows PX4 to store it's parameters on.

### CODE REVIEW

**Comment by TSC21** [comment]
@LorenzMeier FYI

File: arch/arm/src/kinetis/kinetis_flexcan.c [hunk 1]
```diff
@@ -996,7 +1001,10 @@ static void kinetis_txdone(FAR struct kinetis_driver_s *priv, uint32_t flags)
    * new XMIT data
    */
 
+  net_lock();
   devif_poll(&priv->dev, kinetis_txpoll);
+  net_unlock();
+  up_enable_irq(priv->config->mb_irq);
```
Comment (davids5) [inline arch/arm/src/kinetis/kinetis_flexcan.c]: Why are these changes not enabling or disabling the the IRQ from the IP block?

**Comment by PetervdPerk-NXP** [comment]
Any update on this PR? it's required to enable parameter storage for https://github.com/PX4/Firmware/pull/14865

**Review by davids5 — CHANGES_REQUESTED** [review]
@PetervdPerk-NXP are these changes already in upstream?  

We would need then to be up streamed and back ported or on the next upgradw we will loose the changes.

File: arch/arm/src/kinetis/kinetis_flexcan.c (hunk 1 above)
Comment (davids5) [reply arch/arm/src/kinetis/kinetis_flexcan.c]: > Any update on this PR? it's required to enable parameter storage for PX4/Firmware#14865

Sorry I had asked a question days ago but did not hit comment. So it was sitting here.

Comment (PetervdPerk-NXP) [reply arch/arm/src/kinetis/kinetis_flexcan.c]: I took over the idea from the Kinetis Enet driver (Which apparantly isn't ideal :wink:). But you're right I can also use the FlexCAN Interrupt Mask (IMASK1) register. Let me try & test using that register.

Comment (davids5) [reply arch/arm/src/kinetis/kinetis_flexcan.c]: > I took over the idea from the Kinetis Enet driver

yep - I just fixed that!

**Review by davids5 — DISMISSED** [review]
Thank you @PetervdPerk-NXP

**Comment by davids5** [comment]
@dagar do you want to test with PX4/Firmware#14865?

**Comment by PetervdPerk-NXP** [comment]
I've tested it with https://github.com/PX4/Firmware/pull/14865 found a mistake in SocketCAN regarding timestamping. Latest commit fixes that, I will make a seperate PR to backport that change to upstream NuttX

**Comment by davids5** [comment]
@PetervdPerk-NXP - I just merged the upstream version of f76dc9a. Please add [BACKPORT] to the commit message and force push here.

**Comment by davids5** [comment]
Stale 10.0.0+ and backports has brought this in

**Pull request closed without merging**